src/js/thirdparty/utl/primordials.js

```diff
@@ -0,0 +1,214 @@
+const createSafeIterator = (factory, next) => {
+  class SafeIterator {
+    constructor(iterable) {
+      this._iterator = factory(iterable);
+    }
+    next() {
+      return next(this._iterator);
+    }
+    [Symbol.iterator]() {
+      return this;
+    }
+  }
+  Object.setPrototypeOf(SafeIterator.prototype, null);
+  Object.freeze(SafeIterator.prototype);
+  Object.freeze(SafeIterator);
+  return SafeIterator;
+};
+
+function getGetter(cls, getter) {
+  // TODO: __lookupGetter__ is deprecated, but Object.getOwnPropertyDescriptor doesn't work on built-ins like Typed Arrays.
+  return Function.prototype.call.bind(cls.prototype.__lookupGetter__(getter));
+}
+
+function getterCaller(getter) {
+  return val => {
+    return val.constructor.prototype.__lookupGetter__(getter).call(val);
+  };
+}
+
+function uncurryThis(func) {
+  return Function.prototype.call.bind(func);
+}
+
+const copyProps = (src, dest) => {
+  Array.prototype.forEach.call(Reflect.ownKeys(src), key => {
+    if (!Reflect.getOwnPropertyDescriptor(dest, key)) {
+      Reflect.defineProperty(dest, key, Reflect.getOwnPropertyDescriptor(src, key));
+    }
+  });
+};
+
+const makeSafe = (unsafe, safe) => {
```

Comment:
Let's not use any of this library 

We can figure out how to have JSC give us these functions instead of a JS-based primordials implementation